src/jobflow_remote/jobs/jobcontroller.py

```diff
@@ -3570,6 +3576,190 @@ def remove_batch_process(self, process_id: str, worker: str) -> dict:
             upsert=True,
         )
 
+    def delete_job(
+        self,
+        job_id: str | None = None,
+        db_id: str | None = None,
+        job_index: int | None = None,
+        delete_output: bool = False,
+        wait: int | None = None,
+        break_lock: bool = False,
+    ) -> list[int]:
+        """
+        Delete a single job from the queue store and optionally from the job store.
+        The Flow document will be updated accordingly but no consistency check
+        is performed. The Flow may be left in an inconsistent state.
+        For advanced users only.
+
+        Parameters
+        ----------
+        job_id : Optional[str]
+            The uuid of the job to delete.
+        db_id : Optional[str]
+            The db_id of the job to delete.
+        job_index : Optional[int]
+            The index of the job. If None, the job with the largest index will be selected.
+        delete_output : bool, default False
+            If True, also delete the job output from the JobStore.
+        wait : Optional[int]
+            In case the Flow or Job is locked, wait this time (in seconds) for the lock to be released.
+        break_lock : bool, default False
+            Forcibly break the lock on locked documents.
+
+        Returns
+        -------
+        list
+            List of db_ids of the deleted Jobs.
+        """
+
+        job_lock_kwargs = dict(projection=["uuid", "index", "db_id", "state"])
+        # avoid deleting jobs in batch states. It would require additional
+        # specific handling and it is an unlikely use case.
+        with self.lock_job_flow(
+            job_id=job_id,
+            db_id=db_id,
+            job_index=job_index,
+            acceptable_states=DELETABLE_STATES,
+            wait=wait,
+            break_lock=break_lock,
+            job_lock_kwargs=job_lock_kwargs,
+        ) as (job_lock, flow_lock):
+            job_doc = job_lock.locked_document
+            if job_doc is None:
+                raise RuntimeError("No job document found in lock")
+            if flow_lock.locked_document is None:
+                raise RuntimeError("No document found in flow lock")
+
+            # Update FlowDoc
+            flow_doc = FlowDoc.model_validate(flow_lock.locked_document)
+            job_uuid, job_index = job_doc["uuid"], job_doc["index"]
+
+            # Remove job from ids list
+            flow_doc.ids = [
+                id_tuple
+                for id_tuple in flow_doc.ids
+                if id_tuple[1] != job_uuid or id_tuple[2] != job_index
+            ]
+
+            # Remove job from jobs list if no job with that id remains in the flow
+            if not any(job_uuid == id_tuple[1] for id_tuple in flow_doc.ids):
+                flow_doc.jobs.remove(job_uuid)
```

Comment:
Maybe I'm not enough familiar with the structure of the FlowDoc but I thought the jobs list had multiple occurrences of the same job uuid if there has been some replace. If that's the case shouldn't it always remove the job_uuid ? (i.e. it would remove one of them). Then related to that, if it's the case, is there a matching between the jobs list and the ids list ? Maybe I just missed the fact that the jobs attribute in FlowDoc contains the unique uuids.

Reply:
My idea was that the `jobs` attribute should only contain one entry for each uuid, but I see that indeed when a replace is present it contains the same uuid multiple times. Aside from this specific case it should not cause issues, since it should always be used just to find a Flow that contains a Job with a specific uuid. So it does not matter that it is present multuple times. Nonetheless I would consider this a bug and fix it. 
In that case this would be correct, but maybe I will change this to 
```python 
flow_doc.jobs = [jid for jid in flow_doc.jobs if jid != job_uuid]
```
to be backward compatible.
What do you think?